### DISCUSSION TITLE
Creatures upgraded in Hill Fort

### DISCUSSION THREAD

**Discussion by ghost** [ihhub/fheroes2#7462] [Ideas]
How do you feel about the fact that the Barbarian faction is very dependent on the Hill Fort  object ? For example, if orcs will be excluded from the number of creatures that improved in this object, would this comply with the principles described here https://github.com/ihhub/fheroes2/wiki/Beyond-1.0 ?

**Comment by zenseii** [comment]
Hi, @Ben80-hero. As we've written in the linked to page, we really only want to add new buildings, so in this case a building similar to hill fort just not upgrading orks. 

Also what is the goal and the reason for removing just orks from the hill fort? Ogres are also barbarian troops.

**Reply by ghost** [reply]
Hi, @zenseii !

I wrote that the Barbarians are too dependent on the Hill Fort object, but I didn't mean to completely deprive them of the bonuses that this object gives. Depriving the Barbarians of the Ogre upgrade would be too harsh (the Ogre upgrade is far more significant than the Orc upgrade). In addition, Gnomes and Ogres seem to make up a couple - both of them are slow and tenacious creatures. I don't want to ruin this couple.

Regarding buildings - new or old:
Hill Fort is not a building in a castle, it is an object on the map. The article on the link says about buildings, monsters, spells, animations, but does not say about objects on the map. That's why I asked.

**Reply by zenseii** [reply]
I see and thank you for more explanation. Basically we need to update that document so that adventure map objects are included too. Not sure when this document was updated last.

My reason for not touching how the hill fort works is because it was always down to a map maker to make a map that is balanced according to factions, so when they decided to place the Hill fort building they should've thought about this.

If your sentiment is that there is no way to make a map currently in which the hill fort doesn't ruin balance then I suggest discussing a new adventure object that is more balanced. We are not restricted to original buildings, and this should be taken advantage of.

**Comment by ghost** [comment]
Thank you too !

> If your sentiment is that there is no way to make a map currently in which the hill fort doesn't ruin balance then I suggest discussing a new adventure object that is more balanced. We are not restricted to original buildings, and this should be taken advantage of.

Yes, that's exactly what I think.
In fact, I believe that the authors of the game made a huge number of mistakes (in terms of game analytics), which cannot be corrected only by adding something new.
Also for example I could question this one (from here https://github.com/ihhub/fheroes2/wiki/Adding-a-new-option-into-the-game):
"No specific numbers applied to objects, artifacts, creatures and etc. within the game. For example, "Peasants have extra 2 HP" is not acceptable as it rises questions like "why 2 but not 3?" or "how about other monsters?"
If one follow this logic, then it is generally impossible to create any game, since such a question will always arise (why 2, and not 3). Therefore, this logic is wrong.

However, since I can't change the mindset of the development team, I'm willing to participate in other discussions - yes, we can discuss a new object that would be an alternative to Hill Fort. I could create this thread soon, or wait until some developer deems it appropriate to do so at a later date.

**Reply by ihhub** [reply]
Hi @Ben80-hero , as explained by @zenseii , the team takes an approach to replicate the original game as close as possible. Modifying the existing item / object / behavior within the engine can be done only when there is indeed a very solid argument for doing so. Also based on discussions in the past any balancing changes must be done with providing statistical data, not just based on gut feeling or personal preference. If the upgrade in Hill Fort should not exist we need to have a "proof" by doing so. How to do that? Most likely run automated tests with AI playing all maps containing this object and check if this modification changes the balance in the correct direction.

Changing the existing objects would confuse existing players and they would mark this as a bug since they know how the object should behave.

**Comment by ghost** [comment]
>How to do that? Most likely run automated tests with AI playing all maps containing this object and >check if this modification changes the balance in the correct direction.

The idea is good, but not everything is as simple as it seems. When trying to get a result using simulations, we get a large mathematical variance that exceeds the magnitude of the effect under study (to a greater extent due to the fact that the variance is very large). So I would trust classical analytical mathematics more than simulations.
